src/test/java/uk/gov/pay/directdebit/payments/resources/PaymentRequestResourceTest.java

```diff
@@ -0,0 +1,180 @@
+                .body(JSON_CHARGE_KEY, is(notNullValue()))
+                .body(JSON_AMOUNT_KEY, isNumber(AMOUNT))
+                .body(JSON_REFERENCE_KEY, is(expectedReference))
+                .body(JSON_DESCRIPTION_KEY, is(expectedDescription))
+                .body(JSON_RETURN_URL_KEY, is(returnUrl))
+                .contentType(JSON);
+
+        String externalPaymentRequestId = response.extract().path(JSON_CHARGE_KEY).toString();
+        String documentLocation = expectedPaymentRequestLocationFor(accountId, externalPaymentRequestId);
+        String chargeTokenId = app.getDatabaseTestHelper().getTokenByPaymentRequestExternalId(externalPaymentRequestId);
+
+        String hrefNextUrl = "http://Frontend" + FRONTEND_CARD_DETAILS_URL + "/" + chargeTokenId;
+        String hrefNextUrlPost = "http://Frontend" + FRONTEND_CARD_DETAILS_URL;
+
+        response.header("Location", is(documentLocation))
+                .body("links", hasSize(3))
+                .body("links", containsLink("self", "GET", documentLocation))
+                .body("links", containsLink("next_url", "GET", hrefNextUrl))
+                .body("links", containsLink("next_url_post", "POST", hrefNextUrlPost, "application/x-www-form-urlencoded", new HashMap<String, Object>() {{
+                    put("chargeTokenId", chargeTokenId);
+                }}));
+        String requestPath2 = CHARGE_API_PATH
+                .replace("{accountId}", accountId)
+                .replace("{paymentRequestExternalId}", externalPaymentRequestId);
+
+
+        ValidatableResponse getChargeResponse = givenSetup()
+                .get(requestPath2)
+                .then()
+                .statusCode(OK.getStatusCode())
+                .contentType(JSON)
+                .body(JSON_CHARGE_KEY, is(externalPaymentRequestId))
+                .body(JSON_AMOUNT_KEY, isNumber(AMOUNT))
+                .body(JSON_REFERENCE_KEY, is(expectedReference))
+                .body(JSON_DESCRIPTION_KEY, is(expectedDescription))
+                .body(JSON_RETURN_URL_KEY, is(returnUrl));
+
+
+        // Reload the charge token which as it should have changed
+
+        String newChargeTokenId = app.getDatabaseTestHelper().getTokenByPaymentRequestExternalId(externalPaymentRequestId);
+
+        String newHrefNextUrl = "http://Frontend" + FRONTEND_CARD_DETAILS_URL + "/" + newChargeTokenId;
+
+        getChargeResponse
+                .body("links", hasSize(3))
+                .body("links", containsLink("self", "GET", documentLocation))
+                .body("links", containsLink("next_url", "GET", newHrefNextUrl))
+                .body("links", containsLink("next_url_post", "POST", hrefNextUrlPost, "application/x-www-form-urlencoded", new HashMap<String, Object>() {{
+                    put("chargeTokenId", newChargeTokenId);
+                }}));
+
+    }
+
+    @Test
+    public void shouldReturn400IfMandatoryFieldsMissing() {
```

Comment:
With Exception mappers we wouldn't need to test this at integration level and fix issues we have in other projects having too many integration tests and few unit (when it should be the other way around) but can be done in next PR